File: Problems/118_Pascal's_Triangle.cpp
#include<iostream>
#include<vector>
using namespace std;

class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        ans.push_back(vector<int>{1});

        for(int i=1; i<numRows;i++){
            vector<int> new_row;
            new_row.push_back(1);
            for(int j=0; j<ans[ans.size()-1].size()-1; j++){
                new_row.push_back(ans[ans.size()-1][j] + ans[ans.size()-1][j+1]);
            }
            new_row.push_back(1);
            ans.push_back(new_row);
        }
        return ans;
    }
};

int main(){
    Solution sol;
    int numRows = 5;
    sol.generate(numRows);
}

File: Problems/118/118_Pascal's_Triangle.py
from typing import List
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = [[1]]
        for i in range(numRows-1):
            curr = [1]
            for j in range(len(ans[-1])-1):
                    curr.append(ans[-1][j] + ans[-1][j+1])
            curr.append(1)
            ans.append(curr)
        return ans
